pkgs/by-name/co/cosmic-panel/package.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, just
+, pkg-config
+, rust
+, rustPlatform
+, libglvnd
+, libxkbcommon
+, wayland
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "cosmic-panel";
+  version = "unstable-2023-10-30";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = "cosmic-panel";
+    rev = "39add9bead80e54300b6e18f6823a6590b42caad";
+    sha256 = "sha256-8Tw9xCDH9vIaqGm3UN4KVr8sLHOPT0rSMdvANzDEp7Y=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "cosmic-client-toolkit-0.1.0" = "sha256-Gy46o8l8dYxrHaaTMKA+7dyp1DaG15Sehc02XTwZd4s=";
+      "cosmic-config-0.1.0" = "sha256-eynEjV7eTRoOUA1v4Ac0FP2h9KQtIDx32WkY0hR4xig=";
+      "cosmic-notifications-util-0.1.0" = "sha256-F1+Y74JdpehRPTANzERwNVE6Q6n5f5HAFtawLQVMFrA=";
+      "launch-pad-0.1.0" = "sha256-gFtUtrD/cUVpLxPvg6iLxxAK97LTlvI4uLxo06UYIU4=";
+      "smithay-0.3.0" = "sha256-csbBnhQUOhYrzobgdhOocHNcGrSBX4AlxavnGFrznqs=";
+      "smithay-client-toolkit-0.18.0" = "sha256-GhCZ7Eb6q7SwA+NeHSiHwx/Fnrw3R6Zm5N2meMOJ2/4=";
+      "xdg-shell-wrapper-0.1.0" = "sha256-pvaI/joul7jWTdIrPq3PbBcQGMLZLd2rTu1aIwXiZN8=";
+    };
+  };
+
+  nativeBuildInputs = [ just pkg-config ];
+  buildInputs = [ libglvnd libxkbcommon wayland ];
+
+  dontUseJustBuild = true;
+
+  justFlags = [
+    "--set"
+    "prefix"
+    (placeholder "out")
+    "--set"
+    "bin-src"
+    "target/${rust.lib.toRustTargetSpecShort stdenv.hostPlatform}/release/cosmic-panel"
```

Comment:
I find this quite a bit less readable than having "--set", the key, and the value on one line.

Reply:
I formatted via nixpkgs-fmt

Reply:
We have no official formatter, so we don't have to defer to the formatter when it formats code to be less readable than it was before.

Reply:
All other COSMIC packages have this type of formatting, so it's consistent to have it for this too.

Reply:
No they don't? cosmic-applets formats this exactly as it was before, and cosmic-comp is not formatted with nixpkgs-fmt either.

Reply:
cosmic-settings, and all other cosmic prs have this formatting applied.

Reply:
I respect that we have no full authoritative formatting rules, so I'm generally happy to let people format their own code however they want, but that's different to reformatting existing code to make it (IMO) less readable.

Reply:
@alyssais I mean I like it, and I'll probably maintain this fully, and it is more readable to me IMO, I'll most likely be the main maintainer since I'm kind of unofficially heading the COSMIC for NixOS effort? (There a NixOS matrix channel I made & mod in for COSMIC), another thing is I'll use this as my DE in NixOS, dogfooding the packages and finding issues etc.

Reply:
True so what exactly do you want me to amend?

Reply:
The only thing I really don't like about the reformatting, as I said, is that I think it's significantly less readable to have

```nix
    "--set"
    "bin-src"
    "target/${rust.lib.toRustTargetSpecShort stdenv.hostPlatform}/release/cosmic-panel"
```

than it is to have

```nix
    "--set" "bin-src" "target/${rust.lib.toRustTargetSpecShort stdenv.hostPlatform}/release/cosmic-panel"
```